File: src/my_robot_tutorials/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>

bool handle_add_two_ints(rospy_tutorials::AddTwoInts::Request &req, rospy_tutorials::AddTwoInts::Response &res)
{
    int result = req.a + req.b;
    ROS_INFO("%d + %d = %d", (int)req.a, (int)req.b, (int)result);
    res.sum = result;
    return true;
}

int main (int argc, char **argv)
{
    ros::init(argc, argv, "add_two_ints_client");
    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("/add_two_ints");

    rospy_tutorials::AddTwoInts srv;
    srv.request.a = 5;
    srv.request.b = 6;

    if (client.call(srv)) {
        ROS_INFO("Returned sum is %d", (int)srv.response.sum);
    } else {
	ROS_WARN("Service call failed");
    }
}

File: src/my_robot_tutorials/add_two_ints_rust/src/add_two_ints_client.rs
mod common;
use log::info;
use rosrust;
use std::time;

fn main() {
    common::setup_logger();
    rosrust::init("add_two_ints_client");

    rosrust::wait_for_service("/add_two_ints", Some(time::Duration::from_secs(10)))
        .expect("Service not available");

    let client = rosrust::client::<rosrust_msg::rospy_tutorials::AddTwoInts>("/add_two_ints")
        .expect("Failed to create client");

    match client.req(&rosrust_msg::rospy_tutorials::AddTwoIntsReq { a: 5, b: 6 }) {
        Ok(Ok(response)) => info!("Returned sum is {}", response.sum),
        Ok(Err(e)) => {
            info!("Service returned an error: {:?}", e);
        }
        Err(e) => info!("Failed to call service: {:?}", e),
    }
}
